src/solver/impls/adams_bashforth/adams_bashforth.cxx

```diff
@@ -502,27 +533,7 @@ BoutReal AdamsBashforthSolver::take_step(const BoutReal timeIn, const BoutReal d
   // The initial error is 0.0
   BoutReal err = 0.0;
 
-  // Calculate the coefficients for a single step of size dt
-  const auto coefs = get_adams_bashforth_coefficients(timeIn + dt, times, order);
-
-  // Create some storage for the update to the state (i.e. state(timeIn + dt) = current +
-  // full_update).
-  Array<BoutReal> full_update(nlocal);
-
-  // Note we split the work here into initialisation with std::fill
-  // and a separate double loop to calculate the update. This is
-  // to ensure we can operate on the contiguous arrays in history
-  // in order.
-  std::fill(std::begin(full_update), std::end(full_update), 0.0);
-
-  for (int j = 0; j < order; j++) {
-    const BoutReal factor = coefs[j];
-
-    BOUT_OMP(parallel for);
-    for (int i = 0; i < nlocal; i++) {
-      full_update[i] += history[j][i] * factor;
-    }
-  }
+  Array<BoutReal> full_update = AB_integrate(nlocal, timeIn + dt, times, history, order);
```

Comment:
Does this involve a copy or is it likely to be "in-place"?

Reply:
It should be in-place with the majority of compilers, and if not, it will be a cheap move instead